crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -915,11 +922,117 @@ pub mod common_conditions {
     /// app.run(&mut world);
     /// assert_eq!(world.resource::<Counter>().0, 0);
     /// ```
-    pub fn not<Marker, T>(condition: T) -> impl Condition<()>
+    pub fn not<Marker, T>(condition: T) -> ConditionInverter<Marker, T::System>
     where
+        Marker: 'static,
         T: Condition<Marker>,
     {
-        condition.pipe(|In(val): In<bool>| !val)
+        let condition = IntoSystem::into_system(condition);
+        let name = format!("!{}", condition.name());
+        ConditionInverter::<Marker, T::System> {
+            _marker: PhantomData,
+            condition,
+            name: Cow::Owned(name),
+        }
+    }
+}
+
+/// Inverts the output of a [`Condition`].
+///
+/// See [`common_conditions::not`] for examples.
+
+pub struct ConditionInverter<Marker, T>
```

Comment:
I think a better name for this would be `NotSystem`, to stay consistent with types like `FunctionSystem`, `CombinatorSystem`, and `PipeSystem`.  
Also, instead of hard-coding this to use `bool`, it might be useful for some niche users if this was generic over `std::ops::Not` types.